.pre-commit-config.yaml

```diff
@@ -0,0 +1,59 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.0.1
+    hooks:
+      - id: check-added-large-files
+      - id: check-docstring-first
+      - id: check-json
+      - id: check-merge-conflict
+      - id: check-symlinks
+      - id: check-toml
+      - id: check-yaml
+      - id: end-of-file-fixer
+      - id: mixed-line-ending
+      - id: no-commit-to-branch
+      - id: pretty-format-json
+      - id: trailing-whitespace
+-   repo: https://github.com/psf/black
+    rev: 21.5b2
+    hooks:
+      - id: black
+        args: ["-l", "120"]
```

Comment:
Pre-commit hook is fine but I prefer if the settings for black, mypy and friends are committed as settings in the repository so my tools do fight with this pre-commit.

Reply:
I transfered black but unfortunately pycodestyle only supports tox.ini or setup.cfg files... so I added the tox.ini file with only pycodestyle settings to avoid problems with your tools, although it isn't very clean.